WalletWasabi.Tests/UnitTests/WabiSabi/Crypto/OwnershipProofTests.cs

```diff
@@ -0,0 +1,44 @@
+using NBitcoin;
+using WalletWasabi.Crypto;
+using WalletWasabi.Tests.Helpers;
+using Xunit;
+
+namespace WalletWasabi.Tests.UnitTests.WabiSabi.Crypto
+{
+	/// <summary>
+	/// Tests for <see cref="OwnershipProof"/> class.
+	/// </summary>
+	public class OwnershipProofTests
+	{
+		[Fact]
+		public void EqualityTest()
+		{
+			using Key key = new();
+			uint256 roundHash = BitcoinFactory.CreateUint256();
+
+			// Request #1.
+			OwnershipProof request1 = CreateOwnershipProof(key, roundHash);
+
+			//// Request #2.
+			OwnershipProof request2 = CreateOwnershipProof(key, roundHash);
+
+			Assert.Equal(request1, request2);
```

Comment:
There's an implicit dependency here on deterministic nonces in the P2WPKH signature, but NBitcoin's behavior is to grind signatures by default. This behaves correctly because the BIP 322 signing does:

```cs
						key.Sign(hash, new SigningOptions(SigHash.All, useLowR: false)),
```

I did not verify but I suppose `useLowR: false` not just disables grinding, but also makes the signature deterministic.

It might make sense to copypaste lines 16-25 as an independent unit test, or make an analogous one just for the BIP 322 signatures, in case there is ever a regression that would be less confusing to debug. However, I rate the chances of that getting screwed up very low, and this test will fail if that happens, so feel free to ignore.